testsuites/basic/provisioning.py

```diff
@@ -20,18 +18,16 @@ class Provisioning(TestSuite):
         the case fails on any panic in kernel
         """,
         priority=0,
-        requirement=simple_requirement(supported_features=[StartStop]),
     )
-    def restart(self) -> None:
+    def smoke_test(self) -> None:
         node = self.environment.default_node
         dmesg = node.tools[Dmesg]
 
         dmesg.check_kernel_panic()
 
         timer = create_timer()
-        start_stop = node.features[StartStop]
         self.log.info(f"restarting {node.name}")
-        start_stop.restart()
+        node.reboot()
```

Comment:
How does this handle when `reboot` doesn't work, but SSH works? (As is the case with at least one test image.) Where is the platform reboot API?

Reply:
If SSH works, but reboot doesn't work, the case should pass. The smoke test case is very different with other cases, as it needs to run before ssh session ready, and try best to reboot and test again. It's not covered in this PR. Platform reboot doesn't not help a lot in this case. If most commands like reboot doesn't work, it's hard to detect if reboot succeed. We will check if there is such a distro, and worth to test.